pkg/cluster/cluster.go

```diff
@@ -205,17 +205,18 @@ func (c *Cluster) send(ev *clusterEvent) {
 }
 
 func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
-	needDeleteCluster := true
+	clusterFailed := false
 
 	defer func() {
-		if needDeleteCluster {
-			c.logger.Infof("deleting cluster")
+		if clusterFailed {
+			c.reportFailedStatus()
```

Comment:
> a cluster event should never blocks the controller.

This is not only cluster event, but also controller restore case. Waiting all exits only happens on controller restore.

> if reporting fails, then the cluster cannot connect to API server.

Add a comment?